### ISSUE TITLE
install.sh causes symlink to /tmp which breaks npm

### ISSUE THREAD

**Issue by reversefold** [npm/npm#17036]
I'm trying to install npm on a vagrant vm based on debian jessie64. Nodejs is installed with:
```
curl -sL https://deb.nodesource.com/setup_4.x | bash -
apt-get install -f -y  --force-yes -q nodejs
```
Running install.sh says it works but the npm command doesn't work:
```
curl --location --silent https://npmjs.org/install.sh | clean=no /bin/bash
```
```
==> eleven: tar=/bin/tar
==> eleven: version:
==> eleven: tar (GNU tar) 1.27.1
==> eleven: Copyright (C) 2013 Free Software Foundation, Inc.
==> eleven: License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
==> eleven: This is free software: you are free to change and redistribute it.
==> eleven: There is NO WARRANTY, to the extent permitted by law.
==> eleven:
==> eleven: Written by John Gilmore and Jay Fenlason.
==> eleven: install npm@latest
==> eleven: fetching: https://registry.npmjs.org/npm/-/npm-5.0.3.tgz
==> eleven: removed 210 packages in 6.17s
==> eleven: /usr/bin/npm -> /usr/lib/node_modules/npm/bin/npm-cli.js
==> eleven: + npm@5.0.3
==> eleven: added 1 package in 1.427s
==> eleven: It worked
==> eleven: /tmp/vagrant-shell: line 73: npm: command not found
```
If I look at the VM's filesystem, the npm module is symlinked to a temporary directory in /tmp which no longer exists.
```
root@debiancontrib-jessie:~# ls -al /usr/bin/npm
lrwxrwxrwx 1 root root 38 Jun  6 22:44 /usr/bin/npm -> ../lib/node_modules/npm/bin/npm-cli.js
root@debiancontrib-jessie:~# ls -al /usr/lib/node_modules/npm
lrwxrwxrwx 1 root root 30 Jun  6 22:44 /usr/lib/node_modules/npm -> ../../../tmp/npm.21762/package
root@debiancontrib-jessie:~# ls -al /usr/lib/node_modules/../../../tmp/npm.21762
ls: cannot access /usr/lib/node_modules/../../../tmp/npm.21762: No such file or directory
```

**Comment by chrmoritz** [comment]
Looks like this issue is releated to the 2nd issue reported at #16916: `make install` (which is used under the hood by `install.sh`) can't be used to install npm to a prefix anymore, because `make install` is just an alias for `node bin/npm-cli.js install -g -f ${NPMOPTS}` and there was a breaking change in `npm@5`'s `npm install <folder>` to create only a symlink to the folder instead of copying the folder over.

This is in fact nothing you would expect from a standard `make install` (on unix) and it has at least broken third party packagers using `make install` and their official `install.sh` script that way.

**Comment by Nullizer** [comment]
wtf the bot

**Comment by adrianheine** [comment]
Fixed in #17626.
